nav2_smac_planner/src/analytic_expansion.cpp

```diff
@@ -132,6 +132,14 @@ typename AnalyticExpansion<NodeT>::AnalyticExpansionNodes AnalyticExpansion<Node
 
   float d = node->motion_table.state_space->distance(from(), to());
 
+  // If the length is too far, exit. This prevents unsafe shortcutting of paths
+  // into higher cost areas far out from the goal itself, let search to the work of getting
+  // close before the analytic expansion brings it home. This should never be smaller than
+  // 4-5x the minimum turning radius being used, or planning times will begin to spike.
+  if (d > _search_info.analytic_expansion_max_length) {
```

Comment:
```suggestion
  if (d*_collision_checker->getCostmap()->getResolution() > _search_info.analytic_expansion_max_length) {
```
Didn't check if it has changed in recent updates but `d` was in map cell size units instead of meters. In that case default and readme values should be reevaluated accordingly.

Reply:
Yup, sorry mentioned in the ticket https://github.com/ros-planning/navigation2/issues/2659#issuecomment-974570505 but will push commit shortly that changes